packages/components/button/src/button-custom.ts

```diff
@@ -17,9 +17,15 @@ export function useButtonCustomStyle(props: ButtonProps) {
   return computed(() => {
     let styles: Record<string, string> = {}
 
-    const buttonColor = props.color
+    let buttonColor = props.color
 
     if (buttonColor) {
+      const match = (buttonColor as string).match(/var\((.*?)\)/)
+      if (match) {
+        buttonColor = window
+          .getComputedStyle(window.document.documentElement)
+          .getPropertyValue(match[1])
+      }
```

Comment:
Do we need to be compatible with SSR, such as adding `isClient` judgment. 🤔

Reply:
🤔Is this only available on the client side?

Reply:
It doesn't seem necessary, so let's leave it like this for now.